AK/StringFloatingPointConversions.cpp

```diff
@@ -0,0 +1,2279 @@
+    { 279, 709, { 2615987810513347952ull, 0ull } },
+    { 280, 709, { 523197562102669591ull, 0ull } },
+    { 280, 710, { 1046395124205339181ull, 0ull } },
+    { 280, 711, { 2092790248410678362ull, 0ull } },
+    { 281, 711, { 418558049682135673ull, 0ull } },
+    { 281, 712, { 837116099364271345ull, 0ull } },
+    { 281, 713, { 1674232198728542689ull, 0ull } },
+    { 281, 714, { 3348464397457085378ull, 0ull } },
+    { 282, 714, { 669692879491417076ull, 0ull } },
+    { 282, 715, { 1339385758982834152ull, 0ull } },
+    { 282, 716, { 2678771517965668303ull, 0ull } },
+    { 283, 716, { 535754303593133661ull, 0ull } },
+    { 283, 717, { 1071508607186267321ull, 0ull } },
+    { 283, 718, { 2143017214372534642ull, 0ull } },
+    { 284, 718, { 428603442874506929ull, 0ull } },
+    { 284, 719, { 857206885749013857ull, 0ull } },
+    { 284, 720, { 1714413771498027714ull, 0ull } },
+    { 284, 721, { 3428827542996055428ull, 0ull } },
+    { 285, 721, { 685765508599211086ull, 0ull } },
+    { 285, 722, { 1371531017198422171ull, 0ull } },
+    { 285, 723, { 2743062034396844342ull, 0ull } },
+    { 286, 723, { 548612406879368869ull, 0ull } },
+    { 286, 724, { 1097224813758737737ull, 0ull } },
+    { 286, 725, { 2194449627517475474ull, 0ull } },
+    { 287, 725, { 438889925503495095ull, 0ull } },
+    { 287, 726, { 877779851006990190ull, 0ull } },
+    { 287, 727, { 1755559702013980379ull, 0ull } },
+    { 287, 728, { 3511119404027960758ull, 0ull } },
+    { 288, 728, { 702223880805592152ull, 0ull } },
+    { 288, 729, { 1404447761611184303ull, 0ull } },
+    { 288, 730, { 2808895523222368606ull, 0ull } },
+    { 289, 730, { 561779104644473722ull, 0ull } },
+    { 289, 731, { 1123558209288947443ull, 0ull } },
+    { 289, 732, { 2247116418577894885ull, 0ull } },
+    { 290, 732, { 449423283715578977ull, 0ull } },
+    { 290, 733, { 898846567431157954ull, 0ull } },
+    { 290, 734, { 1797693134862315908ull, 0ull } },
+};
+
+// Checks if value is divisible by 5 ^ power.
+bool is_divisible_by_pow_5(u64 value, i32 power)
+{
+    static const u64 FIVE_POWERS[] = { 5, 25, 625, 390625, 152587890625 };
+
+    if (power <= 0 || value == 0) {
+        return true;
+    }
+    if (power >= 28) { // 2 ^ 64 - 1 < 5 ^ 28
+        return false;
+    }
+
+    i32 result = 0;
+    for (i32 i = 5; i--;) {
+        if (value % FIVE_POWERS[i] == 0) {
+            value /= FIVE_POWERS[i];
+            result += 1 << i;
+        }
+    }
+    return result >= power;
+}
+
+template<>
+FloatingPointExponentialForm convert_floating_point_to_fixed_decimal(double value)
+{
+    static const u64 LEN_M = 52;
+    static const i32 BIAS = 1023;
+    static const i32 LOOKUP_OFFSET = BIAS + LEN_M + 1;
+
+    u64 bit_representation = bit_cast<u64, double>(value);
+
+    bool sign = bit_representation >> 63;
+    i32 e = static_cast<i32>(bit_representation >> LEN_M) & ((1 << 11) - 1);
+    u64 m = bit_representation & ((1ull << LEN_M) - 1);
```

Comment:
how about a bitfield union, like LibC/math.cpp uses:
https://github.com/SerenityOS/serenity/blob/db2e1bfa027774ff1e1f883c4ddadc277c1e6e1e/Userland/Libraries/LibC/math.cpp#L85-L98

Reply:
I agree, also we should probably put that somewhere multiple spots can use it, I think (Un)SignedBitInteger also has a copy already

Reply:
AK/FloatingPoint has this I think